### ISSUE TITLE
Problem with installing ASTRA to Ubuntu computer

### ISSUE THREAD

**Issue by 013630700** [astra-toolbox/astra-toolbox#171]
Hi,
for some reason there is a problem with mex files. When I was installing ASTRA, I got this kind of configuration error: Unable to determine matlab mex suffix. 
Everything should be okay, I have MATLAB R2018a and I have the following compilers:

g++                 4:5.3.1-1ubuntu1         amd64        GNU C++ compiler
ii  g++-5           5.4.0-6ubuntu1~16.04.10   amd64        GNU C++ compiler

I also tried to install older ASTRA verision, with this kind of error messages:

checking for mex... /usr/bin/mex
/usr/bin/mex: unrecognized option '-cxx'
/usr/bin/mex: option '-output' is ambiguous; possibilities: '--output-comment' '--output-directory' '--output-format'
This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015/Debian) (preloaded format=mex)
 restricted \write18 enabled.
entering extended mode
! I can't find file `conftest'.
<*> conftest 
             conftest.o
(Press Enter to retry, or Control-D to exit)
Please type another input file name: 
! Emergency stop.
<*> conftest 
             conftest.o
No pages of output.
Transcript written on texput.log.
configure: error: in `/home/sallalat/astra/build/linux':
configure: error: Unable to determine matlab mex suffix
See `config.log' for more details

I would be really thankful if someone could tell me how to fix this issue. Thank you.

**Comment by wjp** [comment]
Hi,

It looks like you're not passing the right path to matlab to astra's configure script. This option looks like `--with-matlab=/usr/local/MATLAB/R2017b`.

**Comment by 013630700** [comment]
Hi, thanks for your rapid answer! I just have the issue, that MATLAB really is located on the path I wrote, it is not in the local folder. Is this a problem?

**Comment by wjp** [comment]
The exact folder doesn't matter, as long as you pass it to astra's configure script. Do you have the `mex` tool installed as part of matlab? It should be in matlab's `bin` directory.

**Comment by 013630700** [comment]
Hi, 
yes, for some reason I did not have mex tool in matlab2018a. It was in 2017b, so I will use that. Problem solved, thanks for everyone, for your help! :)

**Comment by farhi** [comment]
We at Synchrotron Soleil have packaged Astra Toolbox for Debian. All dependencies are solved, and the package has been pushed to Debian SID "unstable" as well as in Ubuntu-like derivatives. The work has been subcontracted to the Freexian company, specialized in packaging. We have considered 'python' and 'octave' bindings, as Matlab is commercial.

You can follow our effort here:
- tracker: https://tracker.debian.org/pkg/astra-toolbox
- SID source package: https://packages.debian.org/source/sid/astra-toolbox
- Binary package: https://packages.debian.org/sid/octave-astra-toolbox
- Binary package: https://packages.debian.org/sid/python3-astra-toolbox

So, in short it will be part of all future Debian-like distributions.
